### DISCUSSION TITLE
Using Supabase and database design

### DISCUSSION THREAD

**Discussion by cspecter** [supabase/supabase#1308] [General]
I am evaluating Supabase as an alternative to our current Firestore setup using the Flamelink CMS. We are also coming from experience with Contentful, DatoCMS and other headless CMS solutions.

We are building a storefront and  would like a tool where we can seed in product information through the CMS. I'm familiar with Postgres, though have been stuck in NoSQL land for a while. I'm curious as to how you recommend we set up our DB and whether can be accomplished within the Supabase dashboard without creating our own CMS. 

This is a simplified version of what I am trying to solve for: We have a list of products, product categories, tags and brands. A product can have 1 brand, 1 category and many tags. 

Setup like this:
```
// tables
- brand

- categories

- tags

- products
-----brand_id // foreign key
-----category_id //foreign key

- product_tags //Join table
-----product_id // foreign key
-----tag_id // foreign key

```

From a data entry perspective I would like to just enter all the info for a new product row, then when picking a relation, pick the Brand, Category and Tags from a list. Instead it seems like I need to look up the ID and manually enter it. Is there no way to pick from all the available IDs in the foreign table? 

Similar question from the join table perspective. I'd love to just pick a list of tags and a product and have it create all the rows automatically? Can this be accomplished in the dashboard? Or will the require a custom rolled CMS solution?

We have several hundred products to seed in and it would be awesome to have a Contentful type solution powered by Postgres.

**Comment by cspecter** [comment]
I answered my own question. We are using [Directus](https://directus.io/) as the CMS to set up and seed our tables. Works perfect!

**Reply by dbredvick** [reply]
Hmm that link 404s for me. Would love to hear more about how you dealt with this.

**Reply by cspecter** [reply]
I updated the link in my original. - https://directus.io/.

It is pretty easy to get Directus set up. We actually used the one-click app on DigitalOcean, but it could really be hosted anywhere. NOTE: If you do use the DO version you will have to SSH in and edit the .env with the correct settings pointing to  Supabase.

Coming from some of the other headless CMS systems out there, like Contentful, Directus seems to have most of the same features. It makes managing relations a little easier (at least from the CMS perspective) pushing a lot of the logic in the background. For our usecase, where we have non-tech editors, I think it will work.

The one thing that will most likely need to be created to make it a full CMS option for Supabase would be a storage adapter into the Supabase Storage system. Right now we are loading files in the Supabase dashboard. It would be nice to get previews in Directus, but not a deal killer right now.

**Reply by dbredvick** [reply]
Ahh yeah bridging the image upload would be very interesting. Seems like it would be doable since Directus is [open source](https://github.com/directus/directus).

**Reply by cspecter** [reply]
Definitely. Might take a look at it. Right now trying to migrate over from Firestore. Lesson learner: don't try to build an app that is highly relational on Firebase. The costs will kill you.

**Reply by silentworks** [reply]
Directus is an awesome CMS, the team over there is also very helpful. I hope someone in the team will help you in getting this setup, If I remember correctly they also have an s3 adapter for their file uploads, so might be some connection here.
